app/src/main/java/com/github/geohunt/app/data/repository/AuthRepository.kt

```diff
@@ -21,16 +21,19 @@ import kotlinx.coroutines.tasks.await
 class AuthRepository(
     private val auth: FirebaseAuth = FirebaseAuth.getInstance(),
     private val authUi: AuthUI = AuthUI.getInstance()
-) {
+): AuthRepositoryInterface {
     /**
-     * Returns the currently authenticated user.
+     * Returns the currently authenticated user as it is stored in Firebase Auth.
      *
      * The caller must first check that there is a user logged in by [requireLoggedIn], otherwise
      * this function will fail with a [NullPointerException].
      *
+     *
      * @return a Firebase user, converted to the external model.
      */
-    fun getCurrentUser(): User {
+    @Deprecated("If you use this function, you very probably want the user as it is" +
+            "defined in the RTDB and not in Firebase Auth. Consider using `UserRepository.getCurrentUser()`.")
```

Comment:
Will there be any real difference between the two ?

As we are approaching the refactor right now, there's minimal info on Firebase RTDB, and they match those of Firebase Auth. Maybe we should discuss what we store in the user on the RTDB ?

Reply:
In my own PR I have been using the repository :aaa:

Reply:
> Will there be any real difference between the two ?
> As we are approaching the refactor right now, there's minimal info on Firebase RTDB, and they match those of Firebase > Auth. Maybe we should discuss what we store in the user on the RTDB ?

Yes, such as the display name (that can be changed on the RTDB). To recall, we can not fetch other users from Firebase Auth without a backend :')